cli/command_benchmark_compression.go

```diff
@@ -106,13 +128,49 @@ func (c *commandBenchmarkCompression) run(ctx context.Context) error {
 		}
 	}
 
-	log(ctx).Infof("Repeating %v times per compression method (total %v). Override with --repeat=N.", repeatCount, units.BytesString(int64(repeatCount*len(data))))
+	algorithms := map[compression.Name]compression.Compressor{}
 
 	for name, comp := range compression.ByName {
-		if compression.IsDeprecated[name] && !c.deprecated {
-			continue
+		if c.shouldIncludeAlgorithm(name) {
+			algorithms[name] = comp
+		}
+	}
+
+	log(ctx).Infof("Will repeat each benchmark %v times per compression method (total %v). Override with --repeat=N.", repeatCount, units.BytesString(int64(repeatCount*len(data))))
+
+	switch c.operations {
+	case "compress":
+		benchmarkCompression = true
+		benchmarkDecompression = false
+	case "decompress":
+		benchmarkCompression = false
+		benchmarkDecompression = true
+	default:
+		benchmarkCompression = true
+		benchmarkDecompression = true
+	}
+
+	if benchmarkCompression {
+		if err := c.runCompression(ctx, data, repeatCount, algorithms); err != nil {
+			return err
 		}
+	}
+
+	if benchmarkDecompression {
+		if err := c.runDecompression(ctx, data, repeatCount, algorithms); err != nil {
```

Comment:
How does this work when multiple algorithms are specified and only decompression is benchmarked? essentially, what's the input for multiple decompressors when there is no prior compression.

Reply:
we compress once per compressor, then decompress N times.

Reply:
I see. I missed that. Thanks.